p4studio/interactive/interactive_command.py

```diff
@@ -166,7 +166,8 @@ def create_default_profile() -> Profile:
 
 
 def build_sde(context: Context, profile: Profile) -> None:
-    plan = ProfileExecutionPlan(profile, None, os.cpu_count())
+    jobs = calculate_jobs_from_available_cpus_and_memory()
```

Comment:
Probably should be an argument from the previous calculation?

Reply:
I am sorry, but I do not understand the comment.  The new first line of `build_sde` calculates and stores a value in variable `jobs`.  The second line calls `ProfileExecutionPlan`, with the 3rd parameter being `jobs` instead of `os.cpu_count()`.

What are you recommending I change?

Reply:
https://github.com/p4lang/open-p4studio/blob/c5c4e982210c1083fa7cb9057f6f8bd48f6fcd16/p4studio/profile/profile_command.py#L215

this calculation is already done here, no? Ideally, we should only do this once and then reuse the result in other contexts.

Reply:
In case it is not obvious: the code in this file is run if you do a command `p4studio interactive`, which is what the `install.sh` bash script invokes.  If you do this, the other Python source file that has the definition of `calculate_jobs_from_available_cpus_and_memory` has none of its code executed at all, unless it is explicitly called from somewhere in this file.  That is what this change is adding -- an explicit call from this file into the function defined in the other file, that calculates a reasonable number of parallel jobs to use.

This is not a repeated call of that function in this code path.  It is the only one.

Reply:
What is the common root between the profile and the interactive command? I was thinking that this denominator should call the function, then pass the result to either one.

Reply:
I think the common ancestor is some `click` module Python code that parses some command line arguments, and decides via click annotations on Python methods which function to invoke.  If you do `p4studio profile <some args>` on the command line, it invokes a method in the file `profile_command.py`.  If you do `p4studio interactive`, it iknvokes a method in this file being modified.

Reply:
Note this line early in this file:
```
from profile.profile_command import execute_plan, calculate_jobs_from_available_cpus_and_memory
```
There is an `execute_plan` function called from both `profile_command.py` and `interactive_command.py` that is doing most of the work of building and installing the software.  Whoever wrote this code decided to define it in `profile_command.py` and call it from here, rather than putting `execute_plan` in some common 3rd file and calling it from both places.  I'm not adding new code flows that didn't exist between files already here.  I'm adding one more cross-file call where there was already a more significant one before.

Reply:
Hmm, I think we can leave it there then. It seems like checks like these should be set from both workflows, but it may be that the interactive command has less restrictions.